management/networksecuritygroup/client.go

```diff
@@ -0,0 +1,235 @@
+// Package networksecuritygroup implements operations for managing network security groups
+// using the Service Management REST API
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913824.aspx
+package networksecuritygroup
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/management"
+)
+
+const (
+	createSecurityGroupURL           = "services/networking/networksecuritygroups"
+	deleteSecurityGroupURL           = "services/networking/networksecuritygroups/%s"
+	getSecurityGroupURL              = "services/networking/networksecuritygroups/%s"
+	listSecurityGroupsURL            = "services/networking/networksecuritygroups"
+	addSecurityGroupToSubnetURL      = "services/virtualnetwork/%s/subnets/%s/networksecuritygroups"
+	getSecurityGroupForSubnetURL     = "services/virtualnetwork/%s/subnets/%s/networksecuritygroups"
+	removeSecurityGroupFromSubnetURL = "services/virtualnetwork/%s/subnets/%s/networksecuritygroups/%s"
+	setSecurityGroupRuleURL          = "services/networksecuritygroups/%s/rules/%s"
+	deleteSecurityGroupRuleURL       = "services/networksecuritygroups/%s/rules/%s"
+
+	errParamNotSpecified = "Parameter %s is not specified."
+)
+
+// NewClient is used to instantiate a new SecurityGroupClient from an Azure client
+func NewClient(client management.Client) *SecurityGroupClient {
+	return &SecurityGroupClient{client: client}
+}
+
+// CreateNetworkSecurityGroup creates a new network security group within
+// the context of the specified subscription
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913818.aspx
+func (sg *SecurityGroupClient) CreateNetworkSecurityGroup(
+	name string,
```

Comment:
it looks like we need a `struct CreateNetworkSecurityGroupParameters` here to follow consistency.